src/index.ts

```diff
@@ -50,12 +50,22 @@ import type {
   Options,
   State,
 } from "./types";
+
+// types required by external middlewares (aws-lambda, etc)
+export type {
+  HandlerOptions,
+  OctokitRequest,
+  OctokitResponse,
+} from "./middleware/types";
+
+// generic handlers
+export { handleRequest } from "./middleware/handle-request";
+
 export { createNodeMiddleware } from "./middleware/node/index";
 export {
   createCloudflareHandler,
   createWebWorkerHandler,
 } from "./middleware/web-worker/index";
-export { createAWSLambdaAPIGatewayV2Handler } from "./middleware/aws-lambda/api-gateway-v2";
```

Comment:
If we are removing exposed methods, should we do an intermediate version deprecating these upcoming removals?

Reply:
Yes, this will be released as a major due to the breaking changes. I was just wondering if we want to do an intermediate minor version warning about the upcoming deprecations to our users.

I guess is more a question to the core maintainers.

Reply:
I agree, that is our usual process. Ideally, a breaking change removes previously deprecated APIs.

Reply:
Also, it would be great if there was a dedicated npm package that would provide the `createAWSLambdaAPIGatewayV2Handler` handler. We could release it as an `@octokit/*` package, but I'd prefer if somone else who actually uses the handler would release it, and we could link to it from the readme and the deprecation message.

Reply:
I would resolve this conversation for the moment and, if a user comes by with the request we can open a Discussion.